docs/advanced/hep_examples.rst

```diff
@@ -255,33 +270,61 @@ For ``ratio`` or ``relative_difference``, the uncertainties can be split between
         hspace=0.3,
     )
 
-    fig_temp, ax_temp = plt.subplots()
-
-    for k_comp in [1, 2, 3, 4]:
-        ratio_uncertainty = "uncorrelated" if k_comp % 2 == 0 else "split"
-        mc_uncertainty = False if k_comp > 2 else True
-
-        fig_comp, ax_main, ax_comparison = compare_data_mc(
+    background_sum = sum(background_hists)
+    
+    fig, ax_main, ax_comparison = compare_data_mc(
             data_hist=data_hist,
             mc_hist_list=background_hists,
             signal_hist=signal_hist,
-            xlabel=key,
+            xlabel="",
             ylabel="Entries",
             mc_labels=background_categories_labels,
             mc_colors=background_categories_colors,
             comparison="ratio",
+            ratio_uncertainty="split",
             fig=fig,
-            ax_main=axes[0] if k_comp == 1 else ax_temp,
-            ax_comparison=axes[k_comp],
+            ax_main=axes[0],
+            ax_comparison=axes[1],
+        )
+    add_text(
+        f'  $\mathbf{{→}}$ comparison = "ratio", \n  $\mathbf{{→}}$ ratio_uncertainty="split", mc_uncertainty = True',
+        ax=ax_comparison,
+        fontsize=10,
+    )
+
+    for k_comp, (ratio_uncertainty, mc_uncertainty) in enumerate([
+        ("uncorrelated", True),
+        ("split", False),
+        ("uncorrelated", False),
+        ], start=2):
+
+        ax_comparison = axes[k_comp]
+
+        background_sum_copy = background_sum.copy()
+        if not mc_uncertainty:
+            background_sum_copy[:] = np.c_[
+            background_sum_copy.values(), np.zeros_like(background_sum_copy.values())
+        ]
+
+        plot_comparison(
+            data_hist,
+            background_sum_copy,
+            ax=ax_comparison,
+            comparison="ratio",
+            xlabel="",
+            h1_label="Data",
+            h2_label="Pred.",
             ratio_uncertainty=ratio_uncertainty,
-            mc_uncertainty=mc_uncertainty,
+            hist_1_uncertainty="asymmetrical",
         )
-        axes[k_comp].set_xlabel("")
         add_text(
             f'  $\mathbf{{→}}$ comparison = "ratio", \n  $\mathbf{{→}}$ ratio_uncertainty="{ratio_uncertainty}", mc_uncertainty = {mc_uncertainty}',
             ax=ax_comparison,
             fontsize=10,
         )
+        set_fitting_ylabel_fontsize(ax_comparison)
+    
+    axes[-1].set_xlabel(key)
 
     fig.savefig("hep_comparisons_ratio_options.svg", bbox_inches="tight")
```

Comment:
Can you add comments on this example, like when we setup the `mc_variances` to 0, or why we use a copy etc... I think this example is complicated but has everything the user needs to do any plots in HEP.